versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/BaseCommitHelper.java

```diff
@@ -230,6 +218,142 @@ static <R, I> R committingOperation(
     }
   }
 
+  void validateNamespacesHaveNoChildren(
+      Set<ContentKey> deletedNamespaces, StoreIndex<CommitOp> headIndex)
+      throws ReferenceConflictException {
+    if (!persist.config().validateNamespaces()) {
+      return;
+    }
+
+    for (ContentKey namespaceKey : deletedNamespaces) {
+      StoreKey storeKey = keyToStoreKey(namespaceKey);
+
+      for (Iterator<StoreIndexElement<CommitOp>> iter = headIndex.iterator(storeKey, null, false);
+          iter.hasNext(); ) {
```

Comment:
We could stop iterating when we hit a key whose prefix truncated to the size of `namespaceKey` is greater than `namespaceKey`.

Reply:
good point!

Reply:
added